lib/saluki-event/src/metric/value/mod.rs

```diff
@@ -0,0 +1,651 @@
+mod iter;
+
+use std::{collections::HashSet, num::NonZeroU64, time::Duration};
+
+use ddsketch_agent::DDSketch;
+use ordered_float::OrderedFloat;
+use smallvec::SmallVec;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+struct TimestampedValue<T> {
+    timestamp: Option<NonZeroU64>,
```

Comment:
It feels a bit odd that `TimestampedValue` might not have a timestamp. Obviously we can get values without timestamps from the DSD protocol, but I don't think we have any use cases right now where a timestamp never gets assigned (and I'm not sure we will?). Is there ever any meaningful variation in how those timestamps eventually get assigned? If not, it seems simplest to remove this variability from the data model and simply assign the timestamp when we receive the metric. If we're using timestamped vs not as an indicator of whether something should be aggregated or not, it seems like we could mark that more explicitly somehow (or maybe we don't need to? How important is it that those points are explicitly not aggregated vs running through the same logic and simply not getting aggregated because they're already independent? I.e. is it an optimization or a semantic difference?)